OBSDOCS-447: improve clarity about limitations when creating alerting rules in user workload monitoring (#67324)
openshift/openshift-docs: modules/monitoring-creating-alerting-rules-for-user-defined-projects.adoc

[id="creating-alerting-rules-for-user-defined-projects_{context}"]
= Creating alerting rules for user-defined projects

> You can create alerting rules for user-defined projects. Those alerting rules will fire alerts based on the values of the chosen metrics.

Reviewer: ```suggestion
You can create alerting rules for user-defined projects. Those alerting rules <report> or <send> alerts based on the values of the chosen metrics.
```

See source above for suggestions on "Fire". Also, note use of future tense here: https://docs.engineering.redhat.com/display/CCSENABLE/Future+tense

Author: Changed to `trigger`

You can create alerting rules for user-defined projects. Those alerting rules will fire alerts based on the values of chosen metrics.

[WARNING]
====

> If you deploy an alerting rule directly on the Prometheus instance in the `openshift-user-workload-monitoring` project by adding the `leaf-prometheus` label to the rule, only those metrics exposed by your user-defined project will be available to use in the alerting rule.

Reviewer 2: `leaf-prometheus` is not label, label with the key `openshift.io/prometheus-rule-evaluation-scope` and value `leaf-prometheus`

Author: @juzhao - Updated per your suggestion. PTAL.

----
$ oc apply -f example-app-alerting-rule.yaml
----

> 

Reviewer 3: I was thinking about merging the two "important", sth like:

"""
[IMPORTANT]
====
A user-defined alerting rule can include metrics for its own project and cluster metrics. You cannot include metrics for another user-defined project.
For example, an alerting rule for the user-defined project `ns1` can have metrics from `ns1` and cluster metrics, such as the CPU and memory metrics. However, the rule cannot include metrics from `ns2`.

You could add the `openshift.io/prometheus-rule-evaluation-scope: leaf-prometheus` label to make the Prometheus instance deployed in the `openshift-user-workload-monitoring` evaluate the alerting rule instead of the thanos ruler, in that case only those metrics exposed by your user-defined project will be available to use in the alerting rule and any alerting rules you create based on cluster metrics MAY not fire.

Additionally, you cannot create alerting rules for the `openshift-*` core {product-title} projects. {product-title} monitoring by default provides a set of alerting rules for these projects.
"""

WDYT?

(then once the two KCS (reduce latency + send recording rules through remote write) are available we can add them as additional resources)

Author: @machine424 - Your suggestion is good but it makes the Important section much too long. This got me thinking that I just needed to create a whole new conceptual section about the behaviors and limitations when someone creates a new rule in UWM. So I added a new section before the procedure rather than putting the info in the middle of the procedure.

Reviewer 3: Good idea! I left some comments.
